app/Models/Beatmapset.php

```diff
@@ -714,13 +745,29 @@ public function disqualifyOrResetNominations(User $user, BeatmapDiscussion $disc
         });
     }
 
-    public function qualify($user)
+    public function qualify(User $user)
     {
         if (!$this->isPending()) {
-            return;
+            throw new InvariantException('cannot qualify a beatmapset not in a pending state.');
         }
 
-        DB::transaction(function () use ($user) {
+        $this->getConnection()->transaction(function () use ($user) {
+            // If beatmapset was previously disqualified, reset the queue timer
+            // if the current nominators are current nominators are different from the previous qualified nominations.
```

Comment:
this comment doesn't read quite right. It doesn't help the second line is longer than the first one. Maybe something like "Reset ... if ... and ..." instead?